src/System.Device.Gpio/Interop/Unix/SafeLineHandle.cs

```diff
@@ -19,8 +19,9 @@ public SafeLineHandle()
 
         protected override bool ReleaseHandle()
         {
-            // Contrary to intuition, this does not invalidate the handle (see comment on declaration)
-            Interop.libgpiod.gpiod_line_release(handle);
+            // Do not actually free anything, the handle is not ours (but rather a memory pointer into an
+            // array owned by the chip)
+            handle = IntPtr.Zero;
```

Comment:
I'm curious here. The doc for `gpiod_line_release` claims something else than this comment. Not implementing ReleaseHandle kills the purpose of using SafeHandle to me. IMO:
- We should keep ReleaseHandle as it was
- Remove ReleaseLock and replace usage with Dispose
- SetPinMode IMO seems to me misimplemented as it calls to ReleaseLock and IMO it should re-acquire the line after that (assuming it's not possible to change direction without first doing so) and replace entry in the dictionary

am I missing something here?

Reply:
FYI AFAIK SafeHandle's dispose makes sure we don't call ReleaseHandle multiple times and zeroes out the handle

Reply:
This whole thing is quite delicate, as we see now. I'm trying to change as little as possible, since everything else works fine. Closing and reopening the line may have other side effects (such as glitches on the line and slower performance), so I don't think we should do that if it can be avoided. 

Unfortunately, the "fix" now broke a bunch of other tests, possibly due to leaked references.

Reply:
Now implemented differently - SaveLineHandle no longer derives from SafeHandle, because it must be released when the chip (in our case the LibGpiodController instance) is released. The whole problem exists because if the chip is released (by the finalizer) before an open line is, the crash occurs as that automatically  frees the memory for all lines, invalidating the handle.